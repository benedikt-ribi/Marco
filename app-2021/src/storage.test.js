/**
 * @jest-environment jsdom
 */

import { describe, expect, test } from "@jest/globals";

describe("WebStorage", () => {
  test("get key/value pairs", () => {
    localStorage.setItem("lastUsername", "John");
    localStorage.setItem("expertMode", "true");

    function* getLocalStorageItems() {
      for (let i = 0; i < localStorage.length; i++) {
        const key = localStorage.key(i);
        yield { key, value: localStorage.getItem(key) };
      }
    }

    const local = [...getLocalStorageItems()];

    expect(local).toMatchSnapshot();
  });

  test("get values in object", () => {
    localStorage.setItem("lastUsername", "John");
    localStorage.setItem("expertMode", "true");

    const local = {};

    for (let i = 0; i < localStorage.length; i++) {
      const key = localStorage.key(i);
      local[key] = localStorage.getItem(key);
    }

    expect(local).toMatchSnapshot();
  });
});
