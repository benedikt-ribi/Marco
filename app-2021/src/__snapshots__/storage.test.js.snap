// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebStorage get key/value pairs 1`] = `
Array [
  Object {
    "key": "lastUsername",
    "value": "John",
  },
  Object {
    "key": "expertMode",
    "value": "true",
  },
]
`;

exports[`WebStorage get values in object 1`] = `
Object {
  "expertMode": "true",
  "lastUsername": "John",
}
`;
